Sync/SyncStateMachine.swift

```diff
@@ -73,10 +73,20 @@ open class SyncStateMachine {
     var stateLabelsSeen = [SyncStateLabel: Bool]()
     var stateLabelSequence = [SyncStateLabel]()
 
+    let stateLabelsIncluded: Set<SyncStateLabel>
```

Comment:
Or even `stateLabelSpace`, since you're constraining the search space while looking for `Ready`.